release-2.1: deps: bump go-libedit

knz opened the pull request:
Picks #31306.

cc @cockroachdb/release

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/31308)
<!-- Reviewable:end -->

knz committed:
deps: bump go-libedit

Release note (build change): CockroachDB can now be built from source
on macOS 10.14 (Mojave).

knz commented:
tfyr
